### ISSUE TITLE
Bug: Indexer Agent calls eth node with Authorisation header even if auth not specified.

### ISSUE THREAD

**Issue by dm3ch** [graphprotocol/indexer#256]
I just have found that indexer-agent sends auth header even if there's no auth data in eth node endpoint url.

I have been running `ghcr.io/graphprotocol/indexer-agent:v0.16.0`
I hadn't got any args when I have been executing indexer agent.
I had got following envs:
```
      INDEXER_AGENT_POSTGRES_DATABASE:            graph-node
      INDEXER_AGENT_POSTGRES_HOST:                postgres-postgresql
      INDEXER_AGENT_INDEX_NODE_IDS:               index_node
      INDEXER_AGENT_GRAPH_NODE_QUERY_ENDPOINT:    http://graphprotocol-node-query:8000/
      INDEXER_AGENT_GRAPH_NODE_ADMIN_ENDPOINT:    http://graphprotocol-node-index:8020/
      INDEXER_AGENT_GRAPH_NODE_STATUS_ENDPOINT:   http://graphprotocol-node-query:8030/graphql
      INDEXER_AGENT_PUBLIC_INDEXER_URL:           http://indexer.domain/
      INDEXER_AGENT_NETWORK_SUBGRAPH_ENDPOINT:    https://gateway.testnet.thegraph.com/network
      INDEXER_AGENT_COLLECT_RECEIPTS_ENDPOINT:    https://gateway.testnet.thegraph.com/collect-receipts
      INDEXER_AGENT_ETHEREUM_NETWORK:             rinkeby
      INDEXER_AGENT_DAI_CONTRACT:                 0x9e7e607afd22906f7da6f1ec8f432d6f244278be
      INDEXER_AGENT_NETWORK_SUBGRAPH_DEPLOYMENT:  QmUBRDENCjFt57vVjFc3D4EQoYqXJffKiyFyuGYB2BN8Wi
      INDEXER_AGENT_ETHEREUM: https://some.host
      INDEXER_AGENT_INDEXER_ADDRESS: <deleted>
      INDEXER_AGENT_MNEMONIC: <deleted>
      INDEXER_AGENT_POSTGRES_PASSWORD: <deleted>
      INDEXER_AGENT_POSTGRES_USERNAME: <deleted>
     
```
     
But in mitmproxy output i had found that it adds auth header anyway.
<img width="1254" alt="Screenshot 2021-06-22 at 18 49 58" src="https://user-images.githubusercontent.com/5025313/122964202-af506e00-d38f-11eb-8735-9033c0f78fba.png">

P.S. As far as I understand I found this problem only because my eth node provider (which doesn't require auth) returns 401 if Auth header exists in request to node without auth. :)
P.S.S. I assume that this problem cause somewhere in source code when auth header adds always, not checking if auth data was supplied

**Comment by dm3ch** [comment]
If I right understand this problem is caused by this line in ether.js - 
https://github.com/ethers-io/ethers.js/blob/ce8f1e4015c0f27bf178238770b1325136e3351a/packages/web/src.ts/index.ts#L149

**Comment by dm3ch** [comment]
Using this change in etherjs library should help as far as I understand.
https://github.com/ethers-io/ethers.js/pull/1697

**Comment by dm3ch** [comment]
I built own image with this problem fixed `dm3ch/graphprotocol-indexer-agent:v0.16.0-auth-fix` using following Dockerfile:
```
FROM ghcr.io/graphprotocol/indexer-agent:v0.16.0
RUN sed -i 's/connection.user\ \!\=\ null\ \&\&\ connection.password\ \!\=\ null/connection.user\ \&\&\ connection.password/g' /opt/indexer/node_modules/@ethersproject/web/lib/index.js
```
